codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java

```diff
@@ -69,16 +69,12 @@ public void setFilename(String filename) {
 
     @Override
     public int hashCode() {
-        return getName().hashCode();
+        return FileUploadUtil.hashCode(this);
     }
 
     @Override
     public boolean equals(Object o) {
-        if (!(o instanceof Attribute)) {
-            return false;
-        }
-        Attribute attribute = (Attribute) o;
-        return getName().equalsIgnoreCase(attribute.getName());
+        return !(o instanceof FileUpload) ? false : FileUploadUtil.equals(this, (FileUpload) o);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Ternary operators that can be simplified with || or && [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ASimplifiedTernary)